src/olympia/editors/helpers.py

```diff
@@ -328,26 +355,52 @@ def render_waiting_time_min(self, row):
         return jinja2.escape(r)
 
     @classmethod
-    def translate_sort_cols(cls, colname):
-        legacy_sorts = {
-            'name': 'addon_name',
-            'age': 'waiting_time_min',
-            'type': 'addon_type_id',
-        }
-        return legacy_sorts.get(colname, colname)
-
-    @classmethod
     def default_order_by(cls):
         return '-waiting_time_min'
 
-    @classmethod
-    def review_url(cls, row):
-        return reverse('editors.review', args=[row.addon_slug])
+
+class EditorAllListTable(SQLTable, ItemStateTable, _EditorTableColumnsHelper):
+    addon_name = tables.Column(verbose_name=_lazy(u'Addon'))
+    version_date = tables.Column(verbose_name=_lazy(u'Last Update'))
+    guid = tables.Column(verbose_name=_lazy(u'GUID'))
+    authors = tables.Column(verbose_name=_lazy(u'Authors'),
+                            sortable=False)
+    last_review = tables.Column(verbose_name=_lazy(u'Last Review'),
+                                sortable=False)
+    flags = tables.Column(verbose_name=_lazy(u'Flags'), sortable=False)
+    platforms = tables.Column(verbose_name=_lazy(u'Platforms'),
+                              sortable=False)
+    additional_info = tables.Column(
+        verbose_name=_lazy(u'Additional'), sortable=False)
 
     class Meta:
-        sortable = True
-        columns = ['addon_name', 'addon_type_id', 'waiting_time_min',
-                   'flags', 'applications', 'additional_info']
+        columns = ['addon_name', 'version_date', 'addon_type_id', 'guid',
+                   'authors', 'flags', 'additional_info', 'last_review']
+
+    def render_guid(self, row):
+        return row.guid
+
+    def render_version_date(self, row):
+        return u'<span>%s</span>' % (row.version_date)
```

Comment:
version_date is literally a date (not user entered) so its safe, and guid will have been validated on upload but I'll add a sanitizing function as suggested to cover the risks anyway.